exercises/brandon_hewer/ex02_oo_basics/src/Shape.h

```diff
@@ -0,0 +1,41 @@
+#ifndef SHAPE_H
+#define SHAPE_H
+
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+
+#include <string>
+#include <sstream>
+
+//===-- Shape.h - Shape definition -------*- C++ -*-==========================//
+//
+//                          2-Dimmensional Shapes
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the declaration of the Shape class, which is the base,
+/// abstract class for all shapes.
+///
+//===----------------------------------------------------------------------===//
+class Shape
+{
+public:
+	/// Returns the perimeter of the shape.
+	virtual double perimeter() const = 0;
+
+	/// Returns the area of the shape.
+	virtual double area() const = 0;
+
+	/// Returns the name of the shape.
+	virtual std::string name() const = 0;
+
+	/// Returns the number of sides of the shape.
+	virtual int numberOfSides() const = 0;
+
+	/// Returns the string representation of the shape.
+	virtual std::string toString() const = 0;
+};
```

Comment:
You are using polymorphism. you require a virtual destructor. See [here](https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors). Please do a bit of research about this. This is easy to get wrong but very important